lib/fabrik/DBManager.js

```diff
@@ -42,86 +41,128 @@ const DB_STATES = {
+      return this.getDbBindInfo()
+        .then(bindInfo => {
+          this.bindInfo = bindInfo;
+          logger.info(`Bind Info retrieved - Connecting to :username:${bindInfo.credentials.username} - DB : ${bindInfo.credentials.dbname} - replicaSet : ${bindInfo.credentials.replicaset}`);
+          return this.initDb(
+            _.assign({}, config.mongodb, {
+              url: bindInfo.credentials.uri
+            }));
+        });
+    });
+  }
+
+  getDbBindInfo() {
+    return Promise.try(() => {
+      if (this.bindInfo) {
+        return this.bindInfo;
       }
-      if (_.get(config, 'mongodb.provision.plan_id') !== undefined) {
-        logger.info(`ServiceFabrik configured to use mongo plan: ${config.mongodb.provision.plan_id}`);
-        const plan = catalog.getPlan(config.mongodb.provision.plan_id);
-        DirectorManager.load(plan)
-          .then(directorManager => {
-            this.directorManager = directorManager;
-            if (config.mongodb.deployment_name) {
-              this.directorManager
-                .getBindingProperty(config.mongodb.deployment_name, CONST.FABRIK_INTERNAL_MONGO_DB.BINDING_ID)
-                .then(bindInfo => {
-                  this.dbCredentials = bindInfo.credentials;
-                  const connectParams = `username:${this.dbCredentials.username} - IPs @[10.11.252.20:27017,10.11.252.21:27017,10.11.252.22:27017]- DB : ${this.dbCredentials.dbname} - replicaSet : ${this.dbCredentials.replicaset}`;
-                  logger.info(`Bind Info retrieved - Connecting to : ${connectParams}`);
-                  return bindInfo.credentials.uri && this.initDb(_.assign({}, config.mongodb, {
-                    url: this.dbCredentials.uri
-                  }));
-                }).catch(ServiceBindingNotFound, (err) => logger.warn('MongoDB binding to ServiceFabrik not found. This generally should not occur. More Info:', err));
-              /**
-               *Could have automatically initiated create/update of DB deployment on start up when no binding/instance is found.
-               *However if broker goes HA then  on start only the master must provision / update the DB. Hence external hooks are
-               *provided for create / update, which can be suitably plugged in as part of post deployment hooks and targetted
-               *specfically to master node. At start up app only tries to bind an existing instance.
-               */
-            } else {
-              logger.error(`mongodb property - config.mongodb.deployment_name is undefined. Deployment name must be defined.`);
-              this.dbState = DB_STATES.NOT_CONFIGURED;
-            }
+      return utils
+        .retry(() => this
+          .directorManager
+          .getBindingProperty(config.mongodb.deployment_name, CONST.FABRIK_INTERNAL_MONGO_DB.BINDING_ID), {
+            maxAttempts: 5,
+            minDelay: 5000,
+            predicate: (err) => !(err instanceof ServiceBindingNotFound)
```

Comment:
Will we be able to log this error ? Or will it be logged by DirectorManager ?

Reply:
This error gets logged in initialize method [here](https://github.com/jagadish-kb/service-fabrik-broker/blob/7aa3fda8089470c5e7f413fd66665fe097802b49/lib/fabrik/DBManager.js#L86)

Reply:
ok.